### PR TITLE
fix(version): use tryparse to parse version

### PR DESCRIPTION
**Pull request by vovikdrg**
Fixing https://github.com/wangkanai/Detection/issues/90

### CODE REVIEW

**Comment by vovikdrg** [comment]
I've build with this fix and pushed version to Azure DevOps private channel. After 8 hours in production no more issues with resolver :)

**Comment by wangkanai** [comment]
#89 Please reference to that the browser parser was an implementation of `System.Version` and i would reverse back to `Wangkanai.Detection.Version` while inheriting IComparable for compatibility.

Maybe you could help out with this issue. Study the #77 Migrate to System.Version from Wangkanai.Detection.Version Pull request.

**Comment by vovikdrg** [comment]
@wangkanai Why to go back? system version is ok, the only issue its better to use to use tryparse and create empty version on fail then fail at all.

**Comment by vovikdrg** [comment]
@wangkanai Or what you are saying you want to revert those changes back? I see  how its better because you split by . and then just try to get every portion. But it still not best way since it would need to use int.tryParse otherwise you end-up having strange versions.

**Comment by wangkanai** [comment]
I agreed on that split is not the best to resolve the version. But have implementing our own Version would give us a better control on how version is parse across all browser, engine, device version. `System.Version` is very limited to comprehend with vase majority of the version string (https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Version.cs). 

What is your idea on the subject?

**Comment by vovikdrg** [comment]
@wangkanai Personally I dont like to invent the weal, specially that here I would not care, if we can't parse version (Which happens from bots, and when request is going via proxy) then we can't parse, most important framework is robust and don't throw exceptions. If you decide to go with custom version then does it worse that effort? because you need to write unittest, you need to fix all bugs and so on. Keep it simple ;)

**Comment by wangkanai** [comment]
> @wangkanai Personally I dont like to invent the weal, specially that here I would not care, if we can't parse version (Which happens from bots, and when request is going via proxy) then we can't parse, most important framework is robust and don't throw exceptions. If you decide to go with custom version then does it worse that effort? because you need to write unittest, you need to fix all bugs and so on. Keep it simple ;)

Its great argument you have got there @vovikdrg

**Comment by vovikdrg** [comment]
@wangkanai  Extension actually it good idea because then we don't need parse to be in base class, so it could be

`stringVersion.ParseVersion()`

**Comment by vovikdrg** [comment]
@wangkanai Look at it when you have time and let me know if you are happy with it :) but I see that extension makes it better :)

File: src/Wangkanai.Detection.Core/DetectionVersionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace Wangkanai.Detection
+{
+    public static class DetectionVersionExtensions
+    {
+        public static Version ToVersion(this string version)
```
[the remark this answers is not in the document]
Comment (wangkanai) [reply src/Wangkanai.Detection.Core/DetectionVersionExtensions.cs]: @vovikdrg do you have a better idea how is this extension method to just the string that has version value?

[the remark this answers is not in the document]
Comment (vovikdrg) [reply src/Wangkanai.Detection.Core/DetectionVersionExtensions.cs]: I see your concern. But to fix this you will need to wrap string version into customversion object and then parse into version :) for me it sounds like spaghetti :)

[the remark this answers is not in the document]
Comment (wangkanai) [reply src/Wangkanai.Detection.Core/DetectionVersionExtensions.cs]: browser detection is nowadays getting more complicate to detect.
You should know the root Render engine first and then device type, and then lastly the browser.

example:
Google Chrome for IOS using [Webkit ](https://en.wikipedia.org/wiki/WebKit) as Engine
Google Chrome version >= 28 for Windows desktop using [Blink](https://en.wikipedia.org/wiki/Blink_(browser_engine)) as  Engine

**Pull request merged**